### DISCUSSION TITLE
Lowercase headers

### DISCUSSION THREAD

**Discussion by max8rr8** [uNetworking/uWebSockets.js#180] [Q&A] [answered]
Why there are lower cased headers

**Comment by ghost** [comment]
The key is lower cased so that you don't have to try looking up all permutations of the key

**Comment by max8rr8** [comment]
Ok, but why when forEach res it returns me lowercase header name that is the main problem

**Comment by ghost** [comment]
Yes they are lower cased. Like explained, they are indeed lower cased

**Comment by ghost** [comment, marked as answer]
I can add that the parser does lowercase the header inplace, so there is no original kept. This is an important optimization.
